modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GmosController.scala

```diff
@@ -0,0 +1,101 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.seqexec.model.dhs.ImageFileId
+import edu.gemini.seqexec.server.SeqexecFailure.Unexpected
+import edu.gemini.spModel.gemini.gmos.GmosCommonType.BuiltinROI
+import squants.Length
+
+import scala.concurrent.duration.Duration
+import scalaz.\/
+
+/**
+  * Created by jluhrs on 8/3/17.
+  */
+trait GmosController {
+  import GmosController._
+
+  type Filter
+  type FPU
+  type GmosStageMode
+  type Disperser
+
+  case class BuiltInFPU(fpu: FPU) extends GmosFPU
```

Comment:
I'm not an expert on path dependent types but this maybe doable if you were to declare `GmosFPU` to take the type of `FPU` as a type param?